iota/filters.py

```diff
@@ -142,3 +143,67 @@ def _apply(self, value):
           'result_type': self.result_type.__name__,
         },
       )
+
+
+class AddressNoChecksum(Trytes):
+  """
+  Validates a sequence as a sequence of trytes.
+  """
+
+  def _apply(self, value):
+    # noinspection PyTypeChecker
+    value =\
+      self._filter(
+        filter_chain = f.Type((binary_type, bytearray, text_type, TryteString)),
+        value = value,
+      ) # type: TrytesCompatible
+
+    if self._has_errors:
+      return None
+
+    # If the incoming value already is an Address then we just make sure it has no checksum
+    if isinstance(value, Address):
+      if value.checksum and not value.is_checksum_valid():
+        return self._invalid_value(
+        value     = value,
+        reason    = self.ADDRESS_BAD_CHECKSUM,
+        exc_info  = True,
+
+        template_vars = {
+          'result_type': self.ADDRESS_BAD_CHECKSUM,
+        },
+      )
+      return Address(value.address)
+
+    # First convert to a generic TryteString, to make sure that the
+    # sequence doesn't contain any invalid characters.
+    try:
+      value = TryteString(value)
+    except ValueError:
+      return self._invalid_value(value, self.CODE_NOT_TRYTES, exc_info=True)
```

Comment:
❔ Since `AddressNoChecksum` (prudently) derives from `Trytes`, I think we might be able to replace a lot of this logic with a call to `super(AddressNoChecksum, self)._apply(value)`.

Reply:
I will have to learn a bit more with how super works and how you combine your own logic in your sub with it, but doesn't sound like it'll be too hard.